grammar.js

```diff
@@ -706,21 +706,36 @@ module.exports = grammar({
     // r.val
     record_value: $ => prec.left('17-17', seq($._expr, '.', alias($.identifier, $.identifier_ref))),
 
-    // [f EXCEPT !.foo[bar].baz = 4, !.bar = 3]
+    // [f EXCEPT !.foo["bar"].baz = 4, !.bar = 3]
     except: $ => seq(
-      '[', $._expr, 'EXCEPT',
-      commaList1(seq('!', $._except_val, '=', $._expr)),
-      ']'
+        '[', 
+        field("expr_to_update", $._expr),
+        'EXCEPT',
+        commaList1($.except_update),
+        ']'
     ),
 
-    // .foo[bar].baz
-    _except_val: $ => repeat1(
-      choice(
-        seq('.', alias($.identifier, $.identifier_ref)),
-        seq('[', commaList1($._expr), ']')
-      )
+    // !.foo["bar"].baz = 4
+    except_update: $ => seq(
+        field("update_specifier", seq('!', $.except_update_specifier)),
```

Comment:
`update_specifier` field seems somewhat superfluous

Reply:
My thinking here is that having an explicit field name makes it more convenient to retrieve the appropriate child node directly (e.g. see [usage](https://github.com/will62794/tla-web/blob/1a916f04483ffd9e74e74393060900ef2e189de9/js/eval.js#L756) in the JS interpreter), rather than needing to [retrieve it by a positional index ](https://github.com/will62794/tla-web/blob/1a916f04483ffd9e74e74393060900ef2e189de9/js/eval.js#L1493-L1494)or by [filtering](https://github.com/will62794/tla-web/blob/1a916f04483ffd9e74e74393060900ef2e189de9/js/eval.js#L757) based on child node types. Will defer to you for final judgement on this, though.

Reply:
That makes sense, keep it in then. Also add whatever other fields are missing, adding fields is not a breaking change.

Reply:
I don't think field names are needed for any other locations in the EXCEPT related nodes. Did you mean to add additional fields in other places?